src/burn/engine/detect.cpp

```diff
@@ -397,11 +398,25 @@ static HRESULT DetectAtomFeedUpdate(
         for (DWORD i = 0; i < pApupChain->cEntries; ++i)
         {
             APPLICATION_UPDATE_ENTRY* pAppUpdateEntry = &pApupChain->rgEntries[i];
+            APPLICATION_UPDATE_ENCLOSURE* pEnclosure = pAppUpdateEntry->rgEnclosures;
 
-            hr = UserExperienceOnDetectUpdate(pUX, pAppUpdateEntry->rgEnclosures ? pAppUpdateEntry->rgEnclosures->wzUrl : NULL, 
-                pAppUpdateEntry->rgEnclosures ? pAppUpdateEntry->rgEnclosures->dw64Size : 0, 
-                pAppUpdateEntry->pVersion, pAppUpdateEntry->wzTitle,
-                pAppUpdateEntry->wzSummary, pAppUpdateEntry->wzContentType, pAppUpdateEntry->wzContent, &fStopProcessingUpdates);
+            if (pEnclosure)
```

Comment:
Shouldn't it check whether there's a hash and/or the value of `digestAlgorithm`?

Reply:
The ternary covers the algorithm but I coerced a null to an empty.